erpnext/hr/doctype/payroll_entry/payroll_entry.py

```diff
@@ -132,20 +137,70 @@ def create_salary_slips(self):
 				"payroll_entry": self.name
 			})
 			if len(emp_list) > 30:
-				frappe.enqueue(create_salary_slips_for_employees, timeout=600, employees=emp_list, args=args)
+				self.queue_action('_create_salary_slips_for_employees', timeout=600, employees=emp_list, args=args)
 			else:
-				create_salary_slips_for_employees(emp_list, args, publish_progress=False)
+				self._create_salary_slips_for_employees(emp_list, args, publish_progress=False)
 				# since this method is called via frm.call this doc needs to be updated manually
 				self.reload()
 
+	def _create_salary_slips_for_employees(self, employees, args, publish_progress=True):
+
+		salary_slips_exists_for = frappe.db.sql_list("""
+		select distinct employee from `tabSalary Slip`
+		where docstatus!= 2 and company = %s
+			and start_date >= %s and end_date <= %s
+			and employee in (%s)
+	""" % ('%s', '%s', '%s', ', '.join(['%s']*len(employees))),
+		[args.company, args.start_date, args.end_date] + employees)
+
+		self.check_permission('write')
+		enumerate(employees)
+		for count, emp in enumerate(employees):
+			if emp not in salary_slips_exists_for:
+				args.update({
+					"doctype": "Salary Slip",
+					"employee": emp
+				})
+				ss = frappe.get_doc(args)
+				ss.insert()
+				if publish_progress:
```

Comment:
move this block outside the `if emp not in salary_slips_exists_for` condition so that progress is always shown for all employees